projects/concurrency/cache.go

```diff
@@ -0,0 +1,124 @@
+package concurrency
+
+import (
+	"math"
+	"time"
+)
+
+type Statistics struct {
+	hitRate             int
+	unReadEntries      int
+	averageHit          float64
+	totalReadsAndWrites int
+}
+
+type cacheEntry[V any] struct {
+	value     V
+	timeStamp time.Time
+	numberOfTimesRead int
+}
+
+type Cache[K comparable, V any] struct {
+	entryLimit      int
+	cache           map[K]cacheEntry[V]
+	unReadEntries   int
```

Comment:
As far as I can tell from the code, this is tracking unread _evicted_ entries (i.e. while an entry is still in the map, it doesn't contribute to this counter) - maybe there's a renaming that could make that more clear?